proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,376 @@
+
+We need to collect and review user feedback to decide on what to work on to
+determine what contributions are of interest. We need to tell users how they can
+provide feedback in the best way for us to take action. We also need to
+strategically and fairly manage the amount of in flight work so that open
+requests do not pile up.
+
+### Pull requests with limited background and context
+
+Many pull requests come as unsolicited changes that often do not correspond to
+an open issue that has been discussed by the maintainers or the community. They
+often end up lingering with no communication from the maintainers to indicate
+the feasibility or appropriateness of the changes. As an example, [`aws-cli`
+pull request 2636](https://github.com/aws/aws-cli/pull/2636)  adds functionality
+to configure MFA. It adds or alters 231 lines of code across five files, and is
+a significant new feature. There is no open issue or discussion about the
+feature other than on the PR. It has no comments from AWS employees, other than
+to indicate that it is a “large” pull request, since 2017. Conversation
+continued for a year on the PR by interested users with no other interaction or
+guidance from AWS.
+
+A idea that has not been vetted by the community and the maintainers is
+difficult to approve; it's hard to have a conversation when it involves an idea
+proposal as well as code changes. It usually ends in an unmerged pull request or
+significant code revision, which is effort that was unnecessarily expended by
+the user and potentially the maintainer. We need a structured process for the
+life cycle of an idea that can culminate in a community contribution. We also
+need to prioritize issues against other requests from the community given the
+effort to contribute for users and review for maintainers.
+
+## Goals
+
+The approach for customer contributions should satisfy the following goals:
+
+1. Users should know how they can help. It should be straightforward for
+   community members to make high quality contributions.
+1. No requests should exist without a response. Communication from maintainers
+   should happen in a timely manner, even (especially!) if that’s saying no. An
+   initial response to open PRs should occur within [TBD] business days.
+1. A contribution should be completed through the efforts of the contributor and
+   maintainers and not be left abandoned. The maintainers are responsible see it
+   through to a conclusion.
+1. We should prioritize the things that people want based on feedback they
+   provide on issues. Users should know how to provide that feedback.
+1. Users should know exactly what to expect when interacting with the
+   maintainers. The next steps should be clearly defined.
+
+## Specification
+ 
+ ![Figure A: Flowchart of contribution guide review
+process.](contrib-guide-high-level.png "Figure A: Flowchart of contribution
+guide review process.")
+
+**Figure A: Flowchart of contribution guide review process.**
```

Comment:
All comments addressed except:

> For Is there an open issue for it? state we should include that the issue needs to be marked as available for contribution.

Since the arrow comes in from the 'available for contribution' queue, I'm wondering how this works best.